### DISCUSSION TITLE
Unable to setup the dev env at local for contribution.

### DISCUSSION THREAD

**Discussion by underscoore** [pyscript/pyscript#1426] [Q&A]
If someone can recognize the screenshot below please help me get the dev env setup. I wanted to contribute to the project.

While running the `make setup` command, it getting failed. refer to the below SS
![image](https://user-images.githubusercontent.com/21233349/234795699-ce19675f-91b5-4745-b115-3ad3835f4e42.png)

**Comment by FabioRosado** [comment]
Hello :wave:

Not entirely sure, but a couple of questions:
- Do you have conda installed in your machine?
- Is your pyscript folder inside OneDrive?
- Do you see a `env` file inside your `pyscriptjs` folder?

**Reply by underscoore** [reply]
Hi @FabioRosado ,

- Conda is already installed on my machine, I have downloaded it via Anaconda, Please refer to the below SS.
- 
![image](https://user-images.githubusercontent.com/21233349/235164800-c8446c56-6729-4c86-b903-0ff01323963d.png)

- Yes it was on one driver but now I have moved it to the C:\
- There is no `env` file inside the `pyscriptjs` folder, but there is one `environment.yml` file is there, Refer to the below SS
- 
![image](https://user-images.githubusercontent.com/21233349/235165268-183632f5-cd93-432c-a094-b6d6c82575b5.png)

Still `make setup` command is not working. :/

**Reply by JeffersGlass** [reply]
I wonder if Windows is doing something odd with relative paths? Maybe try copy-pasting the full conda command out of the output there (the one that starts with `conda env create -p C:/Users ...`, and replace `--file environment.yml` with the full path to environment.yml.

**Comment by underscoore** [comment]
Hi @JeffersGlass , @FabioRosado This issue has been resolved now the local env is set up. 
![image](https://user-images.githubusercontent.com/21233349/235173282-b63637cd-4479-4d0a-a812-352cb9946718.png)

Now, while running `npm run dev` this command getting the below error :/
![image](https://user-images.githubusercontent.com/21233349/235173794-7cecf29d-072a-496b-b87b-c51f3b07ae8f.png)

**Reply by hoodmane** [reply]
I think you should use windows subsystem Linux. We don't currently support Windows builds.

**Reply by JeffersGlass** [reply]
You might try `set NODE_WATCH=1 && node esbuild.mjs` (haven't testing this, but that would be the matching syntax for windows commandline)

**Reply by underscoore** [reply]
![image](https://user-images.githubusercontent.com/21233349/235173794-7cecf29d-072a-496b-b87b-c51f3b07ae8f.png)

I think it is saying that the `NODE_WATCH command is not recognized`, I tried this node package this package is not available.

**Reply by underscoore** [reply]
Can anyone help me!! :/

**Reply by JeffersGlass** [reply]
Assuming you've activated your conda environment in the appropriate shell, the equivalent command for the `dev` script would be:

- Windows Powershell: `set NODE_WATCH=1; node esbuild.mjs` or
- cmd.exe: `set NODE_WATCH=1 && node esbuild.mjs`

If you're building on windows, you'll basically need to the scripts in package.json and adjust them a bit to be windows friendly. The listing/formatting commands similarly will need their escaping tweaked to work on windows.

**Comment by underscoore** [comment]
Hi @hoodmane , I have no idea how I can use Windows subsystem Linux, although I have installed Ubuntu using `wsl --install'
Is it possible to connect somewhere, let me know.

**Reply by underscoore** [reply]
Hi, Is there anything I should try?
